lib/foreman/force_ssl.rb

```diff
@@ -0,0 +1,49 @@
+module Foreman
+  class ForceSsl
+    def self.unattended_paths
+      @unattended_paths ||= [%r{/unattended/}, %r{/userdata/}]
+    end
+
+    def self.unattended_paths_regex
+      @unattended_paths_regex ||= Regexp.union(unattended_paths)
+    end
```

Comment:
Can't this just be a constant?
```suggestion
    UNATTENTED_PATHS = %r{^(/(unattended|userdata)/}
```

Reply:
It can, but I wanted to allow plugins to add to that array :)
Tho they can add to the regex :shrug:

Reply:
I'm not sure if we should allow that. In the plugins I have checked out I'm not sure I see a valid use case. I'm slightly leaning to the other direction: push all client contact to the Smart Proxy and make Foreman pure SSL. If we do that, we could even implement the HTTP -> HTTPS redirect in Apache instead, which saves serving requests via Ruby.

Reply:
I agree, it's definitelly a better direction, I'll take a look into plugins that use this today and see how hard would that be to implement :)

Reply:
I believe we use use this (i.e. an opt out of force-ssl) in a couple of plugins, I'd need to check which ones though.

Reply:
@laugmanuel: Do you have time to check this?

Reply:
@timogoebel, I had a short look and did not see any usage of this in our plugins.

Reply:
@laugmanuel are you sure you're not using the `smart_proxy_auth`?

Reply:
@ezr-ondrej I can't find any traces of `smart_proxy_auth` in our foreman plugins. I hope, I didn't miss anything.

Reply:
I wouldn't expect it to use `smart_proxy_auth` but rather `SmartProxyAuth` as a concern.